src/parser/parser.h

```diff
@@ -0,0 +1,112 @@
+ * LinescaleGUI is free software: you can redistribute it and/or modify       *
+ * it under the terms of the GNU General Public License as published by       *
+ * the Free Software Foundation, either version 3 of the License, or          *
+ * (at your option) any later version.                                        *
+ *                                                                            *
+ * LinescaleGUI is distributed in the hope that it will be useful,            *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the               *
+ * GNU General Public License for more details.                               *
+ *                                                                            *
+ * You should have received a copy of the GNU General Public License          *
+ * along with linescaleGUI. If not, see <http://www.gnu.org/licenses/>.       *
+ ******************************************************************************/
+/**
+ * @file parser.h
+ * @authors Gschwind, Weber, Schoch, Niederberger
+ *
+ * @brief Parses the line scale communication protocol
+ *
+ */
+
+#ifndef PARSER_H
+#define PARSER_H
+
+#include <QByteArray>
+#include <QObject>
+
+/**
+ * @brief Indicates the working mode sent by the Line Scale
+ *
+ */
+enum class WorkingMode {
+    REALTIME,      ///< Indicates that the work is in real-time mode
+    OVERLOADED,    ///< Indicates that the test is overloaded
+    MAX_CAPACITY,  ///< Indicates the maximum capacity (default 3000 not received)
+};
+/**
+ * @brief   Indicates if the measured force is relative to a previous set value or if it is absolute
+ *
+ */
+enum class MeasureMode {
+    ABS_ZERO,  ///< Represents the absolute zero measurement mode
+    REL_ZERO,  ///< Represents the relative zero measurement mode
+};
+/**
+ * @brief Unit of the measured force
+ *
+ */
+enum class UnitValue {
+    KN,   ///< Indicates the unit of measurement is kN
+    KGF,  ///< The unit of measurement is kgf
+    LBF,  ///< Indicates that the unit of measurement is lbf
+};
+
+/**
+ * @brief holds parsed data
+ *
+ */
+struct Sample {
+    WorkingMode workingMode; ///< Indicates the working mode sent by the Line Scale
+    double measuredValue;   ///< Stores value of measured force
+    MeasureMode measureMode;  ///< Indicates if the measured force is relative to a previous set value or if it is absolute
+    double referenceZero;   ///< Stores reference force
+    int batteryPercent; ///< Stores battery voltage of the Line Scale in percent
+    UnitValue unitValue; ///< Stores the unit of the measured force
+    int frequency; ///< Stores the connection frequency between host device and Line Scale
+};
+
+class Parser : public QObject {
+    Q_OBJECT
+   public:
+    /**
+     * @brief Parses received messages from Line Scale to usable data for GUI.
+     *
+     * @param package Raw package as QByteArray
```

Comment:
```suggestion
     * @brief Parses received messages from Line Scale to usable data for GUI.
     *
     * @todo Return error code instead of bool.
     *
     * @param package Raw package as QByteArray
```